app/Http/Controllers/IncomingTextController.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\TwilioRequest as Request;
+
+class IncomingTextController extends Controller
+{
+    protected $commands = [
+        "unsubscribe" => [
+            "STOP",
+            "STOPAll",
+            "UNSUBSCRIBE",
+            "CANCEL",
+            "END",
+            "QUIT",
+        ],
+        "subscribe" => [
+            "START",
+            "YES",
+        ]
+    ];
+
+    /**
+     * Starting point for all incoming text messages.
+     */
+    public function receiveText(Request $request)
+    {
+        if ($command = $this->translateCommand($request->Body)) {
+            return $this->{$command}($request);
+        }
+
+        return $this->responseMessage('Unrecognized command: '.$request->Body);
+    }
+
+    protected function subscribe(Request $request)
+    {
+        $request->phoneNumber()->markBlacklisted(false);
+    }
+
+    protected function unsubscribe(Request $request)
+    {
+        $request->phoneNumber()->markBlacklisted();
+    }
+
+    /**
+     * Transalate an incoming text to a command using $this->commands.
+     *
+     * @param  string $body The body of the incoming message
+     * @return string|false
+     */
+    protected function translateCommand($body)
+    {
```

Comment:
Wonder if this could be:

```php
protected function translateCommand($body)
{
    return collect($this->commands)->filter(function ($aliases) use ($body) {
        return in_array(strtoupper($body), $aliases);
    })->map(function ($value, $key) {
        return $key;
    })->first();
}
```

I wonder if there's a bit cleaner way than what I showed there, but I haven't had a chance to actually run it. It doesn' treturn `false`, might need restructuring, gotta run to a meeting